### PR TITLE
Manually filter old android device log messages instead of relying on the '-T' option

### PR DESCRIPTION
**Pull request by johnmccutchan**
Fixes Galaxy Note 5 (https://github.com/flutter/flutter/issues/6280)

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@jason-simmons

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -771,7 +772,28 @@ class _AdbLogReader extends DeviceLogReader {
...
+    final Match timeMatch = AndroidDevice._timeRegExp.firstMatch(line);
+    if (timeMatch == null) {
+      return;
+    }
+    if (_timeOrigin != null) {
+      final String timestamp = timeMatch.group(0);
+      DateTime time = _adbTimestampToDateTime(timestamp);
+      if (time.isBefore(_timeOrigin)) {
+        // Ignore log messages before the origin.
+        printTrace('skipped old log line: $line');
```
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Do these need to be printed?  Users can see them at any time by running logcat

File: packages/flutter_tools/lib/src/hot.dart [hunk 2]
```diff
@@ -249,7 +249,13 @@ class HotRunner extends ResidentRunner {
     }
 
     _observatoryPort = result.observatoryPort;
-    await connectToServiceProtocol(_observatoryPort);
+    try {
```
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/hot.dart]: Move this to a separate PR?

**Comment by jason-simmons** [comment]
LGTM

File: packages/flutter_tools/lib/src/hot.dart (hunk 2 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/hot.dart]: This fixes the error message when we fall over detecting the service protocol port.

File: packages/flutter_tools/lib/src/android/android_device.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/android/android_device.dart]: I'm just being consistent with the existing behaviour: log messages that are ignored are still printed via printTrace. See the bottom of this function.

Comment (Hixie) [inline packages/flutter_tools/lib/src/android/android_device.dart]: I added this yesterday. May have been a mistake. Feel free to remove that.
